yarn-project/circuit-types/src/l2_block.ts

```diff
@@ -190,184 +63,38 @@ export class L2Block {
-      this.newCommitments.length,
-      this.newCommitments,
-      this.newNullifiers.length,
-      this.newNullifiers,
-      this.newPublicDataWrites.length,
-      this.newPublicDataWrites,
-      this.newL2ToL1Msgs.length,
-      this.newL2ToL1Msgs,
-      this.newContracts.length,
-      this.newContracts,
-      this.newContractData,
-      this.newL1ToL2Messages.length,
-      this.newL1ToL2Messages,
-      this.newEncryptedLogs,
-      this.newUnencryptedLogs,
-    );
-  }
-
-  /**
-   * Serializes a block without logs to a string.
-   * @remarks This is used when the block is being served via JSON-RPC because the logs are expected to be served
-   * separately.
-   * @returns A serialized L2 block without logs.
-   */
-  toString(): string {
-    return this.toBuffer().toString(STRING_ENCODING);
-  }
-
-  /**
-   * Deserializes L2 block without logs from a buffer.
-   * @param buf - A serialized L2 block.
-   * @returns Deserialized L2 block.
+   * Deserializes a block from a buffer
+   * @returns A deserialized L2 block.
    */
   static fromBuffer(buf: Buffer | BufferReader) {
     const reader = BufferReader.asReader(buf);
     const header = reader.readObject(Header);
     const archive = reader.readObject(AppendOnlyTreeSnapshot);
-    const newCommitments = reader.readVector(Fr);
-    const newNullifiers = reader.readVector(Fr);
-    const newPublicDataWrites = reader.readVector(PublicDataWrite);
-    const newL2ToL1Msgs = reader.readVector(Fr);
-    const newContracts = reader.readVector(Fr);
-    const newContractData = reader.readArray(newContracts.length, ContractData);
-    // TODO(sean): could an optimization of this be that it is encoded such that zeros are assumed
-    const newL1ToL2Messages = reader.readVector(Fr);
+    const body = reader.readObject(Body);
 
     return L2Block.fromFields({
       archive,
       header,
-      newCommitments,
-      newNullifiers,
-      newPublicDataWrites,
-      newL2ToL1Msgs,
-      newContracts,
-      newContractData,
-      newL1ToL2Messages,
+      body,
     });
   }
 
   /**
-   * Deserializes L2 block with logs from a buffer.
-   * @param buf - A serialized L2 block.
-   * @returns Deserialized L2 block.
+   * Serializes a block
+   * @remarks This can be used specifying no logs, which is used when the block is being served via JSON-RPC because the logs are expected to be served
+   * separately.
+   * @returns A serialized L2 block logs.
    */
-  static fromBufferWithLogs(buf: Buffer | BufferReader) {
-    const reader = BufferReader.asReader(buf);
-    const block = L2Block.fromBuffer(reader);
-    const newEncryptedLogs = reader.readObject(L2BlockL2Logs);
-    const newUnencryptedLogs = reader.readObject(L2BlockL2Logs);
-
-    block.attachLogs(newEncryptedLogs, LogType.ENCRYPTED);
-    block.attachLogs(newUnencryptedLogs, LogType.UNENCRYPTED);
-
-    return block;
+  toBuffer() {
+    return serializeToBuffer(this.header, this.archive, this.body.toBuffer());
```

Comment:
You don't have to call .toBuffer on the body here as the serializeToBuffer function does that.